src/components/landing/WaitListForm.tsx

```diff
@@ -73,20 +74,33 @@ const WaitlistForm: React.FC = () => {
 
     setSubmitting(true)
     try {
-      await fetch('/api/join-waitlist', {
+      const response = await fetch('/api/join-waitlist', {
         method: 'POST',
         headers: {
           'Content-Type': 'application/json',
         },
         body: JSON.stringify({
-          formData,
+          data: formData,
         }),
       }).then((res) => res.json())
 
+      console.log('RESPONSE:', response)
+
+      if (response.status && response.message && response.status === 'error') {
+        setFormError(response.message)
+      } else {
+        setSubmitted(true)
+        setFormData({
+          company: '',
+          website: '',
+          firstname: '',
+          email: '',
+        })
+      }
+
       setSubmitting(false)
-      setSubmitted(true)
     } catch (error: any) {
-      setErrors(error.response.data)
+      setFormError('There was an error submitting the form')
```

Comment:
By setting form error instead of errors you are no longer handling individual field related error messages from the server.